webpack/assets/javascripts/react_app/components/EditableTextDisplay/EditableTextDisplay.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Spinner,
+  TextInput,
+  TextArea,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, PencilAltIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../common/I18n';
+import './editableTextDisplay.scss';
+
+const EditableTextDisplay = ({ onEdit, value, textArea, attribute }) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  // Setting didCancel to avoid actions from happening after component has been unmounted
+  // see https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions
+  useEffect(() => {
+    let didCancel = false;
+
+    const onSubmit = async () => {
+      if (submitting) {
+        await onEdit(inputValue, attribute);
+        if (!didCancel) {
+          setSubmitting(false);
+          setEditing(false);
+        }
+      }
+    };
+    onSubmit();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [submitting]);
+
+  // Listen for enter and trigger submit workflow on enter
+  useEffect(() => {
+    const listener = event => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        if (editing) setSubmitting(true);
+      }
+    };
+    document.addEventListener('keydown', listener);
```

Comment:
I think this is just two ways to solve the same problem managing state with an event listener:
1. Add and remove the event listener when user switches to and from editing
2. Use a ref to store the value and adding the event listener on mount

I am open to the ref solution, but I don't fully understand the issue

>This occurs multiple times - on every editing state change.

Can you elaborate on the concern here? Adding and removing the event listener when the user switches editing state shouldn't be a costly operation.

Reply:
I think the concern is:

1. `editing` changes to true - `document.addEventListener` runs
2. `editing` changes to false - `document.addEventListener` runs and a new event listener is added
3. `editing` changes again - a third event listener is created

The cleanup function in the `return` wouldn't run when `editing` changes; it would only run when the component is unmounted (removed from the DOM).

Reply:
> * `editing` changes to false - `document.addEventListener` runs and a new event listener is added

From my understanding [here](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Multiple_identical_event_listeners) any duplicate event listeners are ignored:

>If multiple identical EventListeners are registered on the same EventTarget with the same parameters, the duplicate instances are discarded. They do not cause the EventListener to be called twice, and they do not need to be removed manually with the removeEventListener() method.

I revisited this and remembered why I came to this approach. I'm mostly following [the example](https://patternfly-react.surge.sh/components/text-input#select-text-using-ref) here. In our case, the ref is not attached on mount because the TextInput is not showing. The editing flag will be false and only the Text component will show. You can check this by setting the array to empty and `console.log(inputRef)` at the top of the hook.

When the editing flag changes, the `TextArea` is rendered and the ref is passed as a prop, thats when the ref is now available.

I changed this to remove the event listener if editing is false and the ref is present, I think that more accurately reflects what we want to happen, rather than handling on unmount.

I think having the component dynamically change based on the editing state makes this a tad more complicated than a basic eventListener example, so that is why we have to add and remove the event listener dynamically as well.

Reply:
ok, I came up with what I think is a much simpler approach :yin_yang: 

I moved the inputs to a separate component and the ref there, that way we can do the much simpler attach event listener on mount and remove on unmount as that smaller component will be mounted and unmounted when editing toggles in the parent component.